### PR TITLE
Assignment 5

### PR DESCRIPTION
**Pull request by codrutzu**

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{
+  "launch": {
+
+    "configurations": [],
+    "compounds": [],
+
+  },
+  "ruby.rubocop.executePath": "",
+  "ruby.format": "rubocop" // use rubocop for formatting
```
Comment (florinionce) [inline .vscode/settings.json:9]: If you want to keep the in your repo, it's better to use [editorconfig](https://editorconfig.org/). The vscode specific settings should go in the gitignore

File: Gemfile [hunk 2]
```diff
@@ -26,7 +26,12 @@ gem 'bcrypt', '~> 3.1', '>= 3.1.15'
 
 gem 'cloudinary'
 
-gem 'rails-controller-testing'
+gem 'faker', '2.11.0'
+
+
```
Comment (florinionce) [inline Gemfile:31]: extra space

File: app/controllers/application_controller.rb [hunk 3]
```diff
@@ -1,3 +1,20 @@
 class ApplicationController < ActionController::Base
-  include SessionsHelper, UsersHelper
+  include UsersHelper
+  include SessionsHelper
+
+  def redirect
+    # if params[:preview] == 1 && current_user&.admin?
```
Comment (florinionce) [inline app/controllers/application_controller.rb:6]: do you still need this?

File: app/controllers/cart_products_controller.rb [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+  def create
+    cart = current_user.cart
+    if (cart_product = CartProduct.find_by(product_id: params[:id], cart_id: params[:cart]))
+      cart_product.update_columns(quantity: cart_product.quantity + 1)
+    else
+      cart.products << Product.find(params[:id])
+    end
+    if !params[:id].nil? && params[:route] == 'cart'
+      redirect_to '/cart'
```
Comment (florinionce) [inline app/controllers/cart_products_controller.rb:12]: I would probably use the helper path for this one :D

File: app/controllers/cart_products_controller.rb [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+  end
+
+  def destroy
+    cart_product = CartProduct.find_by(product_id: params[:id])
+    if CartProduct.find_by(product_id: params[:id], cart_id: params[:cart]).quantity > 1
+      cart_product.update_columns(quantity: cart_product.quantity - 1)
+    else
+      CartProduct.delete(cart_product)
+    end
+    redirect_to '/cart'
```
Comment (florinionce) [inline app/controllers/cart_products_controller.rb:25]: same here

File: app/views/user_mailer/account_activation.text.haml [hunk 6]
```diff
@@ -0,0 +1,8 @@
+-# UserMailer#account_activation
+Hi = @user.name
+Welcome to the Eureka caffe! Click on the link below to activate your account:
+
+-# = @greeting + ", find me in app/views/user_mailer_mailer/account_activation.text.haml"
+
+= edit_account_activation_url(@user.activation_token, email: @user.email)
+
```
Comment (florinionce) [inline app/views/user_mailer/account_activation.text.haml:8]: extra space here :D

File: .vscode/settings.json (hunk 1 above)
Comment (codrutzu) [reply .vscode/settings.json:9]: I have some trouble with rubocop, so I should remove those line anyways

File: app/controllers/products_controller.rb [hunk 7]
```diff
@@ -1,23 +1,23 @@
...
   end
 
-  def show 
+  def show
     @product = Product.find(params[:id])
   end
 
-  def create 
+  def create
     @product = Product.new(product_params)
```
Comment (florinionce) [inline app/controllers/products_controller.rb:11]: Can anyone add products or only the admins?

Comment (codrutzu) [reply app/controllers/products_controller.rb:11]: already modified it :D

**Pull request merged**